src/main/java/org/candlepin/subscriptions/tally/facts/FactNormalizer.java

```diff
@@ -206,9 +206,27 @@ private void normalizeSystemProfileFacts(
       normalizedFacts.setCores(
           hostFacts.getSystemProfileCoresPerSocket() * hostFacts.getSystemProfileSockets());
     }
+    if ("x86_64".equals(hostFacts.getSystemProfileArch())
+        && HardwareMeasurementType.VIRTUAL
+            .toString()
+            .equalsIgnoreCase(hostFacts.getSystemProfileInfrastructureType())) {
+      var effectiveCores = calculateVirtualCPU(hostFacts);
+      normalizedFacts.setCores(effectiveCores);
+    }
     getProductsFromProductIds(normalizedFacts, hostFacts.getSystemProfileProductIds());
   }
 
+  private Integer calculateVirtualCPU(InventoryHostFacts virtualFacts) {
+    //  For x86, guests: if we know the number of threads per core and its greater than one,
```

Comment:
About finding the threads per core count, this is future work right ? Phase 2 of this @lindseyburnett @kahowell

Reply:
Yes, using threads per core is future work.

Reply:
@san7ket yup, that's future work